cmd/kubeadm/app/master/manifests.go

```diff
@@ -57,30 +58,38 @@ const (
 // WriteStaticPodManifests builds manifest objects based on user provided configuration and then dumps it to disk
 // where kubelet will pick and schedule them.
 func WriteStaticPodManifests(s *kubeadmapi.KubeadmConfig) error {
+	// Placeholder for kube-apiserver pod spec command
+	apiServerCommand := getComponentCommand(apiServer, s)
+
+	// Check if the user decided to use an external etcd cluster
+	if len(s.InitFlags.API.Etcd.ExternalEndpoints) > 0 {
+		arg := fmt.Sprintf("--etcd-servers=%s", strings.Join(s.InitFlags.API.Etcd.ExternalEndpoints, ","))
+		apiServerCommand = append(apiServerCommand, arg)
+	} else {
+		apiServerCommand = append(apiServerCommand, "--etcd-servers=http://127.0.0.1:2379")
+	}
+
+	// Is etcd secured?
+	if s.InitFlags.API.Etcd.ExternalCAFile != "" {
+		etcdCAFileArg := fmt.Sprintf("--etcd-cafile=%s", s.InitFlags.API.Etcd.ExternalCAFile)
+		apiServerCommand = append(apiServerCommand, etcdCAFileArg)
+	}
+	if s.InitFlags.API.Etcd.ExternalCertFile != "" && s.InitFlags.API.Etcd.ExternalKeyFile != "" {
```

Comment:
This is good for now, but we should probably error or warn if user has only provided one of these.